app/controllers/UtilityController.java

```diff
@@ -3,38 +3,80 @@
 import com.google.common.collect.ImmutableMap;
 import com.typesafe.config.Config;
 import helpers.JsonHelper;
+import interfaces.DocumentStore;
+import interfaces.PdfGenerator;
+import lombok.val;
 import org.joda.time.DateTime;
-import play.libs.Json;
+import play.Logger;
+import play.mvc.Controller;
+import play.mvc.Result;
+
+import javax.inject.Inject;
 import java.io.File;
 import java.net.InetAddress;
 import java.net.UnknownHostException;
 import java.util.Arrays;
 import java.util.HashMap;
-import javax.inject.Inject;
-import lombok.val;
-import play.mvc.Controller;
-import play.mvc.Result;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Stream;
+
+import static java.lang.Runtime.getRuntime;
 
 public class UtilityController extends Controller {
 
     private final String version;
 
+    private PdfGenerator pdfGenerator;
+    private DocumentStore documentStore;
+
     @Inject
-    public UtilityController(Config configuration) {
+    public UtilityController(Config configuration, PdfGenerator pdfGenerator, DocumentStore documentStore) {
 
         version = configuration.getString("app.version");
+        this.pdfGenerator = pdfGenerator;
+        this.documentStore = documentStore;
     }
 
-    public Result healthcheck() {
+    public CompletionStage<Result> healthcheck() {
+        val pdfGeneratorHealthFuture = pdfGenerator.isHealthy().toCompletableFuture();
+        val documentGeneratorHealthFuture = documentStore.isHealthy().toCompletableFuture();
+        val allHealthFutures = CompletableFuture.allOf(pdfGeneratorHealthFuture, documentGeneratorHealthFuture);
 
+        return allHealthFutures
+            .thenApply((ignored) -> buildResult(pdfGeneratorHealthFuture.join(), documentGeneratorHealthFuture.join()));
+    }
+
+    private Result buildResult(Boolean pdfGeneratorStatus, Boolean documentStoreStatus) {
+        return JsonHelper.okJson(new HashMap<String, Object>() {
+        {
+            put("status", pdfGeneratorStatus && documentStoreStatus ? "OK" : "FAILED");
+            put("dateTime", DateTime.now().toString());
+            put("version", version);
+            put("runtime", runtimeInfo());
+            put("fileSystems", fileSystemDetails());
+            put("localHost", localhost());
+            put("dependencies", new HashMap<String, Object>() {
```

Comment:
Could use  ImmutableMap.of() here

Reply:
It only accepts up to 4 entries. Could Use ImmutableMap's builder pattern maybe

Reply:
Converted to the ImmutableMap builder pattern which looks nicer